lib/datadog/grok/benches/filters/keyvalue.rs

```diff
@@ -0,0 +1,77 @@
+use std::time::Duration;
+
+use bytes::Bytes;
+use criterion::{
+    criterion_group, measurement::WallTime, BatchSize, BenchmarkGroup, Criterion, SamplingMode,
+};
+use datadog_grok::filters::keyvalue::{apply_filter, KeyValueFilter};
+use regex::Regex;
+use value::Value;
+
+fn apply_filter_bench(c: &mut Criterion) {
+    let mut group: BenchmarkGroup<WallTime> =
+        c.benchmark_group("datadog_grok::filters::keyvalue::apply_filter");
+    group.sampling_mode(SamplingMode::Auto);
+
+    group.bench_function("apply_filter key=valueStr", move |b| {
+        b.iter_batched(
+            || {
+                let value = Value::Bytes(Bytes::from("key=valueStr"));
+                let filter = KeyValueFilter {
+                    key_value_delimiter: "=".into(),
+                    value_re: Regex::new(r"^[\w.\-_@]+").unwrap(),
+                    quotes: vec![('"', '"'), ('\'', '\''), ('<', '>')],
+                    field_delimiters: [" ", ",", ";"]
+                        .iter()
+                        .map(|x| String::from(*x))
+                        .collect::<Vec<String>>(),
+                };
+                (value, filter)
+            },
+            |(value, filter): (Value, KeyValueFilter)| {
+                let _ = apply_filter(&value, &filter);
```

Comment:
Should we use `black_box` here or return the value so Criterion handles it?

Reply:
I think this whole closure might be automatically black boxed? https://github.com/bheisler/criterion.rs/blob/master/book/src/faq.md#when-should-i-use-criterionblack_box

Reply:
Hmm, isn't:

>  If you're writing an un-parameterized benchmark of a function that takes an argument, however, this may be worth considering.

What we are doing here though?

Reply:
Oh. There used to be language about things being auto black-boxed that was more aggressive. I'll add the black box.

Ultimately this one of those pain points with micro-benchmarks I guess. In practice we want the optimizer to remove as much as possible, except not in this case, but that ends up deviating the results more from what we'd see in practice at a macro level.

Reply:
Hmm, yeah, that is a tension with benchmarks that I never considered before.